packages/http/src/decorators.ts

```diff
@@ -596,6 +596,18 @@ export function $route(context: DecoratorContext, entity: Type, path: string, pa
   });
 }
 
+/**
+ * `@sharedRoute` marks the operation as sharing a route path with other operations.
+ *
+ * When an operation is marked with `@sharedRoute`, it enables other operations to share the same
+ * route path as long as those operations are also marked with `@sharedRoute`.
+ *
+ * `@sharedRoute` can only be applied directly to operations.
+ */
+export function $sharedRoute(context: DecoratorContext, entity: Operation) {
```

Comment:
We should have a decorator definition in TypoeSpec here: https://github.com/microsoft/typespec/blob/main/packages/http/lib/http-decorators.tsp

Reply:
I'll add one, I was surprised to see that `@route` doesn't have one so I'll add one for it too.

Reply:
Done.